app/frontend/src/routes.ts

```diff
@@ -44,24 +44,55 @@ export const routeToSearch = (query: string) => `${searchRoute}/${query}`;
 export const jailRoute = "/restricted";
 export const tosRoute = "/tos";
 
-export const placeRoute = "/place";
+const placeBaseRoute = "/place";
+export const placeRoute = `${placeBaseRoute}/:pageId/:pageSlug?`;
 export const routeToPlace = (id: number, slug: string) =>
   `${placeRoute}/${id}/${slug}`;
 export const newPlaceRoute = `${placeRoute}/new`;
 
-export const guideRoute = "/guide";
+const guideBaseRoute = "/guide";
+export const guideRoute = `${guideBaseRoute}/:pageId/:pageSlug?`;
 export const routeToGuide = (id: number, slug: string) =>
   `${guideRoute}/${id}/${slug}`;
 export const newGuideRoute = `${guideRoute}/new`;
 
-export const communityRoute = "/community";
-export const routeToCommunity = (id: number, slug: string) =>
-  `${communityRoute}/${id}/${slug}`;
-
-export const groupRoute = "/group";
+const groupBaseRoute = "/group";
+export const groupRoute = `${groupBaseRoute}/:groupId/:groupSlug?`;
 export const routeToGroup = (id: number, slug: string) =>
   `${groupRoute}/${id}/${slug}`;
 
-export const discussionRoute = "/discussion";
+const discussionBaseRoute = "/discussion";
+export const discussionRoute = `${discussionBaseRoute}/:discussionId/:discussionSlug?`;
 export const routeToDiscussion = (id: number, slug: string) =>
   `${discussionRoute}/${id}/${slug}`;
+
+const eventBaseRoute = "/event";
+export const eventRoute = `${eventBaseRoute}/:eventId/:eventSlug?`;
+export const routeToEvent = (id: number, slug: string) =>
+  `${eventRoute}/${id}/${slug}`;
+
+const communityBaseRoute = "/community";
+export const communityRoute = `${communityBaseRoute}/:communityId/:communitySlug?`;
+export const routeToCommunity = (id: number, slug: string) =>
+  `${communityRoute}/${id}/${slug}`;
+
+const communityPlacesBaseRoute = `${communityRoute}/places`;
+export const communityPlacesRoute = `${communityPlacesBaseRoute}/:communityId/:communitySlug?`;
+export const routeToCommunityPlaces = (id: number, slug: string) =>
+  `${communityPlacesRoute}/${id}/${slug}`;
```

Comment:
```suggestion
  `${communityPlacesBaseRoute}/${id}/${slug}`;
```
Otherwise redirect is broken. Same for all the dynamic community related route creators here